File: programmers/level1/하샤드_수/10006_하샤드수.java
class Solution {
    public boolean solution(int x) {
        boolean answer = true;
        int result = 0;
        
        String[] aList = String.valueOf(x).split("");
        for (String a : aList) {
            result += Integer.parseInt(a);
        }
        return (x%result == 0);
    }
}

// 가장 인상적인 코드 : int a = (int) char-48;
// int > String > toCharArray() -> char[] > (int) char-48 > int  (GOOD!!!)
// 
// public class HarshadNumber{
//     public boolean isHarshad(int num){
//         char[] charArr = String.valueOf(num).toCharArray();
//         int divisor = 0;
// 
//         for (char c : charArr) {
//             divisor += (int) c - 48;
//         }
//     return (num % divisor == 0);
//     }
// }

File: programmers/level1/하샤드_수/10006_하샤드수.js
function solution(x) {    
    let h = 0;
    x.toString().split("").forEach(e => h+= parseInt(e) );
    
    // let h = x.toString().split("").reduce((a,b) => +a + +b);
    return x%h === 0;
}

// 가장 인상깊은 코드 
// 잊었던 reduce의 상기 & Number + "" = String & +String = Number
// 
// function Harshad(n){
//   return !(n % (n + "").split("").reduce((a, b) => +b + +a ));
// }

